[Failing test] UT TestUnsafeConversions is failing with master golang

### Which jobs are failing?

UT `k8s.io/apiserver/pkg/authentication/token: cache  TestUnsafeConversions ` is failing with following error when running with master golang, Here is the job link, https://prow.ppc64le-cloud.cis.ibm.net/view/gs/ppc64le-kubernetes/logs/postsubmit-master-golang-kubernetes-unit-test-ppc64le/1891896113523331072

```
--- FAIL: TestUnsafeConversions (0.00s)
    --- FAIL: TestUnsafeConversions/toString_allocations (0.00s)
panic: testing: AllocsPerRun called during parallel test [recovered, reraised]

goroutine 34 [running]:
testing.tRunner.func1.2({0x7ce2a0, 0xa1ba50})
	/usr/local/go/src/testing/testing.go:1740 +0x1d8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1743 +0x348
panic({0x7ce2a0?, 0xa1ba50?})
	/usr/local/go/src/runtime/panic.go:804 +0x154
testing.AllocsPerRun(0x64?, 0xc0000f9f00?)
	/usr/local/go/src/testing/allocs.go:22 +0x1cc
k8s.io/apiserver/pkg/authentication/token/cache.TestUnsafeConversions.func4(0xc00023e000)
	/root/kubernetes/staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go:621 +0xe8
testing.tRunner(0xc00023e000, 0xc0001f7920)
	/usr/local/go/src/testing/testing.go:1798 +0x108
created by testing.(*T).Run in goroutine 13
	/usr/local/go/src/testing/testing.go:1857 +0x3ec
FAIL	k8s.io/apiserver/pkg/authentication/token/cache	0.009s
FAIL
```

This started happening post merge of https://go-review.googlesource.com/c/go/+/630137 where `panic` is introduced if test is running in parallel.

### Which tests are failing?

`TestUnsafeConversions`

### Since when has it been failing?

`Feb 15, 2025`

### Testgrid link

_No response_

### Reason for failure (if possible)

With addition of https://go-review.googlesource.com/c/go/+/630137 in Golang, it is required now that test shouldn't run in parallel when `AllocsPerRun` is used as `AllocsPerRun's` result will be flaky.

### Anything else we need to know?

I am able to fix the test by removing parallel executions at places where `AllocsPerRun` is used. Here is the stress tool o/p with the fix,

```
5s: 32 runs so far, 0 failures, 8 active
10s: 64 runs so far, 0 failures, 8 active
15s: 104 runs so far, 0 failures, 8 active
20s: 137 runs so far, 0 failures, 8 active
25s: 176 runs so far, 0 failures, 8 active
30s: 216 runs so far, 0 failures, 8 active
35s: 252 runs so far, 0 failures, 8 active
40s: 289 runs so far, 0 failures, 8 active
45s: 328 runs so far, 0 failures, 8 active
50s: 368 runs so far, 0 failures, 8 active
55s: 403 runs so far, 0 failures, 8 active
1m0s: 441 runs so far, 0 failures, 8 active
1m5s: 481 runs so far, 0 failures, 8 active
1m10s: 520 runs so far, 0 failures, 8 active
1m15s: 556 runs so far, 0 failures, 8 active
1m20s: 593 runs so far, 0 failures, 8 active
1m25s: 633 runs so far, 0 failures, 8 active
1m30s: 672 runs so far, 0 failures, 8 active
1m35s: 708 runs so far, 0 failures, 8 active
1m40s: 745 runs so far, 0 failures, 8 active
1m45s: 785 runs so far, 0 failures, 8 active
1m50s: 824 runs so far, 0 failures, 8 active
1m55s: 860 runs so far, 0 failures, 8 active
2m0s: 897 runs so far, 0 failures, 8 active
2m5s: 937 runs so far, 0 failures, 8 active
2m10s: 976 runs so far, 0 failures, 8 active
2m15s: 1012 runs so far, 0 failures, 8 active
2m20s: 1049 runs so far, 0 failures, 8 active
2m25s: 1089 runs so far, 0 failures, 8 active
2m30s: 1128 runs so far, 0 failures, 8 active
2m35s: 1164 runs so far, 0 failures, 8 active
2m40s: 1202 runs so far, 0 failures, 8 active
2m45s: 1241 runs so far, 0 failures, 8 active
```

### Relevant SIG(s)

/sig api-machinery

--- a/staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go
@@ -552,7 +552,6 @@ func withAudit(ctx context.Context) context.Context {
 }
 
 func TestUnsafeConversions(t *testing.T) {
-	t.Parallel()
 
 	// needs to be large to force allocations so we pick a random value between [1024, 2048]
 	size := utilrand.IntnRange(1024, 2048+1)
@@ -574,7 +573,6 @@ func TestUnsafeConversions(t *testing.T) {
 	})
 
 	t.Run("toBytes allocations", func(t *testing.T) {
-		t.Parallel()
 
 		s := utilrand.String(size)
 		f := func() {
@@ -606,7 +604,6 @@ func TestUnsafeConversions(t *testing.T) {
 	})
 
 	t.Run("toString allocations", func(t *testing.T) {
-		t.Parallel()
 
 		b := make([]byte, size)
 		if _, err := rand.Read(b); err != nil {

